tools/data_loader.py

```diff
@@ -0,0 +1,378 @@
+                                "group_ids": group_ids
+                            },
+                            token=tokens[token_name]
+                        )
+                        telescope_dict[nickname] = data["data"]["id"]
+                if src.get("instruments") is not None:
+                    with status("Creating instruments"):
+                        instrument_dict = {}
+                        ins_src = src.get("instruments")
+                        if isinstance(ins_src, dict) and \
+                                src["instruments"].get("file") is not None:
+                            # we're asked to load a file containing telescopes
+                            fname = str(topdir / src["instruments"]["file"])
+                            ins_src = yaml.load(open(fname, "r"), Loader=Loader)
+                            ins_token = src["instruments"]["token"]
+
+                        for i in ins_src:
+                            if i.get("token") is not None:
+                                token_name = i["token"]
+                            else:
+                                token_name = ins_token
+
+                            if i.get("telescope_nickname") is not None:
+                                telname = i["telescope_nickname"]
+                            else:
+                                telname = i.get("telescope_id")
+                            telid = telescope_dict[telname]
+                            # print(telid, telname, token_name, tokens[token_name], group_ids)
+                            data = assert_post(
+                                "instrument",
+                                data={
+                                    "name": i["name"],
+                                    "type": i["type"],
+                                    "band": i["band"],
+                                    "telescope_id": telid,
+                                    "filters": i.get("filters", [])
+                                },
+                                token=tokens[token_name]
+                            )
+                            instrument_dict[i["name"]] = data["data"]["id"]
+            if src.get("sources") is not None:
+                with status("Loading Source and Candidates"):
+                    (basedir / "static/thumbnails").mkdir(parents=True, exist_ok=True)
+                    for s in src.get("sources"):
+                        sinfo = [{"id": s["id"], "ra": s["ra"],
+                                  "dec": s["dec"], "save_source": True,
+                                  "redshift": s.get("redshift", 0.0),
+                                  "altdata": s.get("altdata", None),
+                                  "cand_filts": s["candidate"]["candidate_filters"],
+                                  "comments": s.get("comments", [])}]
+                        if s.get("unsaved_candidate_copy") is not None:
```

Comment:
Is this general enough to hard-code like this?

Reply:
yes